### ISSUE TITLE
Installation of tflite-model-maker 0.3.2. takes forever

### ISSUE THREAD

**Issue by Extremesarova** [tensorflow/tensorflow#51031]
<em>Please make sure that this is a build/installation issue. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:build_template</em>

**System information**
- OS Platform and Distribution: Windows 10 Enterprise
- TensorFlow installed from (source or binary): from PyPi
- TensorFlow version: 2.5.0
- Python version: 3.6.8 and 3.8.8
- Installed using virtualenv? pip? conda?:  using pip in venv
- Bazel version (if compiling from source): N/A
- GCC/Compiler version (if compiling from source): N/A
- CUDA/cuDNN version: CUDA 11.2, cuDNN 8.1.0
- GPU model and memory: NVIDIA GeForce RTX 2070 with Max-Q Design 8gb

**Describe the problem**
Installation of tflite-model-maker 0.3.2. takes forever.
Attached log demonstrates situation for a 10+ minutes of installation. I will attach fuller log in the comments.
It downloads 17 versions of scipy package, 10+ versions of scikit-learn, 3 versions of tensorflow (although 2.5.0 version is already installed) and so on.
I tried installation of tflite-model-maker 0.3.2 few times, but couldn't make it. 
Installed nightly version, but I question its quality and stability - that's why I've decided to create an issue. 

**Provide the exact sequence of commands / steps that you executed before running into the problem**
pip install tflite-model-maker==0.3.2

**Any other info / logs**
[pip install tflite-model-maker.txt](https://github.com/tensorflow/tensorflow/files/6900929/pip.install.tflite-model-maker.txt)

**Comment by saikumarchalla** [comment]
@Extremesarova  I can able to install it properly and it took less than a minute. Please check the [colab](https://colab.research.google.com/gist/saikumarchalla/ca054239bda81200ac1c532a1c8cc253/untitled117.ipynb) gist. Thanks!

**Comment by Extremesarova** [comment]
Thank you for fast response. I've tried installation under Ubuntu also right now - same as on Windows. 
Attaching updated logs from installation which started over 30 minutes ago and still continues (and on Ubuntu too).
[pip install tflite-model-maker_updated.txt](https://github.com/tensorflow/tensorflow/files/6901076/pip.install.tflite-model-maker_updated.txt)

Can you check on a local machine?

**Comment by Extremesarova** [comment]
The same situation is happening even on a freshly created venv

**Comment by Extremesarova** [comment]
In Colab python version is 3.7.11. Maybe it influences the installation process somehow.

**Comment by Engineero** [comment]
I'm having the same issue. It seems like pip is not able to resolve the dependency tree and downloads/checks many versions of most packages, including building a ton of wheels. I've tried this on Ubuntu and Mac OS. After about 18 hours I finally got an error on Ubuntu (Mac is still running). Traceback below:

```bash
ERROR: Exception:
Traceback (most recent call last):
  File "/home/[username]/.local/lib/python3.6/site-packages/pip/_internal/cli/base_command.py", line 173, in _main
    status = self.run(options, args)
  File "/home/[username]/.local/lib/python3.6/site-packages/pip/_internal/cli/req_command.py", line 203, in wrapper
    return func(self, options, args)
  File "/home/[username]/.local/lib/python3.6/site-packages/pip/_internal/commands/install.py", line 316, in run
    reqs, check_supported_wheels=not options.target_dir
  File "/home/[username]/.local/lib/python3.6/site-packages/pip/_internal/resolution/resolvelib/resolver.py", line 95, in resolve
    collected.requirements, max_rounds=try_to_avoid_resolution_too_deep
  File "/home/[username]/.local/lib/python3.6/site-packages/pip/_vendor/resolvelib/resolvers.py", line 472, in resolve
    state = resolution.resolve(requirements, max_rounds=max_rounds)
  File "/home/[username]/.local/lib/python3.6/site-packages/pip/_vendor/resolvelib/resolvers.py", line 383, in resolve
    raise ResolutionTooDeep(max_rounds)
pip._vendor.resolvelib.resolvers.ResolutionTooDeep: 2000000
```

An example of how it is trying to resolve dependencies. I've never seen pip do this before, but it's downloading every version that seems like it might work and trying them all, often building wheels, although in this example I think it downloads built modules. Note this is my second attempt, so most of these were cached already.

```bash
INFO: pip is looking at multiple versions of scikit-learn to determine which version is compatible with other requirements. This could take a while.
Collecting scikit-learn!=0.19.0,>=0.14.0
  Using cached scikit_learn-0.24.2-cp36-cp36m-manylinux2010_x86_64.whl (22.2 MB)
  Downloading scikit_learn-0.24.1-cp36-cp36m-manylinux2010_x86_64.whl (22.2 MB)
     |████████████████████████████████| 22.2 MB 2.2 MB/s
  Using cached scikit_learn-0.24.0-cp36-cp36m-manylinux2010_x86_64.whl (22.2 MB)
  Using cached scikit_learn-0.23.2-cp36-cp36m-manylinux1_x86_64.whl (6.8 MB)
  Using cached scikit_learn-0.23.1-cp36-cp36m-manylinux1_x86_64.whl (6.8 MB)
  Using cached scikit_learn-0.23.0-cp36-cp36m-manylinux1_x86_64.whl (7.3 MB)
  Using cached scikit_learn-0.22.2.post1-cp36-cp36m-manylinux1_x86_64.whl (7.1 MB)
INFO: pip is looking at multiple versions of scikit-learn to determine which version is compatible with other requirements. This could take a while.
  Using cached scikit_learn-0.22.2-cp36-cp36m-manylinux1_x86_64.whl (7.1 MB)
  Using cached scikit_learn-0.22.1-cp36-cp36m-manylinux1_x86_64.whl (7.0 MB)
  Using cached scikit_learn-0.22-cp36-cp36m-manylinux1_x86_64.whl (7.0 MB)
  Using cached scikit_learn-0.21.3-cp36-cp36m-manylinux1_x86_64.whl (6.7 MB)
  Using cached scikit_learn-0.21.2-cp36-cp36m-manylinux1_x86_64.whl (6.7 MB)
INFO: This is taking longer than usual. You might need to provide the dependency resolver with stricter constraints to reduce runtime. If you want to abort this run, you can press Ctrl + C to do so. To improve how pip performs, tell us what happened here: https://pip.pypa.io/surveys/backtracking
  Using cached scikit_learn-0.21.1-cp36-cp36m-manylinux1_x86_64.whl (6.7 MB)
  Using cached scikit_learn-0.21.0-cp36-cp36m-manylinux1_x86_64.whl (6.6 MB)
  Using cached scikit_learn-0.20.4-cp36-cp36m-manylinux1_x86_64.whl (5.4 MB)
  Using cached scikit_learn-0.20.3-cp36-cp36m-manylinux1_x86_64.whl (5.4 MB)
  Using cached scikit_learn-0.20.2-cp36-cp36m-manylinux1_x86_64.whl (5.4 MB)
  Using cached scikit_learn-0.20.1-cp36-cp36m-manylinux1_x86_64.whl (5.4 MB)
  Using cached scikit_learn-0.20.0-cp36-cp36m-manylinux1_x86_64.whl (5.3 MB)
  Using cached scikit_learn-0.19.2-cp36-cp36m-manylinux1_x86_64.whl (4.9 MB)
  Using cached scikit_learn-0.19.1-cp36-cp36m-manylinux1_x86_64.whl (12.4 MB)
  Using cached scikit_learn-0.18.2-cp36-cp36m-manylinux1_x86_64.whl (11.8 MB)
  Using cached scikit_learn-0.18.1-cp36-cp36m-manylinux1_x86_64.whl (11.8 MB)
  Using cached scikit-learn-0.18.tar.gz (8.9 MB)
  Using cached scikit-learn-0.17.1.tar.gz (7.9 MB)
  Using cached scikit-learn-0.17.tar.gz (7.8 MB)
  Using cached scikit-learn-0.16.1.tar.gz (7.3 MB)
  Using cached scikit-learn-0.16.0.tar.gz (7.3 MB)
  Using cached scikit-learn-0.15.2.tar.gz (7.0 MB)
  Using cached scikit-learn-0.15.1.tar.gz (7.0 MB)
  Using cached scikit-learn-0.15.0.tar.gz (7.0 MB)
  Using cached scikit-learn-0.14.1.tar.gz (6.8 MB)
  Using cached scikit-learn-0.14.tar.gz (6.8 MB)
```

**Comment by Engineero** [comment]
I was able to get it to work. Steps to fix:

1. Copy the [requirements.txt](https://github.com/tensorflow/examples/blob/master/tensorflow_examples/lite/model_maker/requirements.txt) file from the repo into my working directory and installing those requirements explicitly with

    `$ pip install -r requirements.txt` 

2. Install tflite-model-maker with

    `$ pip install tflite-model-maker`

There seems to be a bug right now where tflite-model-maker is not aware of its own requirements (or something funky with pip is happening) and it is trying to resolve them algorithmically instead of just reading the requirements file.

**Comment by smilingday** [comment]
Thanks for reporting and @lintian06 to take a look.

**Comment by lintian06** [comment]
Thank you @Extremesarova for reporting the issue, and @Engineero for the solution!

This is a mysterious issue! I think it might be some conflicts among (1) pre-installed pypi packages, and (2) dependencies required by Model Maker, (may also related to platforms OS + python version). Unfortunately, pip is not good enough to tell the information for us to find the root cause. :-(

We have an automatic CI/CD pipelines to capture pip install failures, but cannot cover all cases. Happy to know if you find the root cause, and I can fit it. (Welcome pull requests too!)

**Comment by Extremesarova** [comment]
> it might be some conflicts among (1) pre-installed pypi packages

I tried installing tflite-model-maker on a clean venv and the error was still there

**Comment by mohantym** [comment]
Hi @Extremesarova ! 
We are checking to see whether you still need help in this issue . Have you checked these threads yet?  Ref [1](https://www.tensorflow.org/lite/guide/model_maker) , [2 ](https://colab.sandbox.google.com/github/tensorflow/tensorflow/blob/master/tensorflow/lite/g3doc/tutorials/model_maker_object_detection.ipynb#scrollTo=sr3q-gvm3cI8), [3](https://colab.sandbox.google.com/gist/mohantym/7cee04a31addf6a5cd4caa8b5b354df2/model-maker-image-classification-tutorial.ipynb) .Thank you!

**Comment by Extremesarova** [comment]
Hi, @mohantym! I will try it and post an update here

**Comment by mohantym** [comment]
Ok @Extremesarova ! Did the issue get resolved ?

**Comment by Huxwell** [comment]
27.04.2023 - the error occurs again, even on Colab:
![image](https://user-images.githubusercontent.com/9086268/234978476-b991f9b7-ab9e-4ca3-9883-e375a7962968.png)
this takes 20min + and doesn't seem to finish soon

**Comment by WENDGOUNDI** [comment]
@Huxwell @YannickFuereder . I have also encounter the same issue in Google Colab and working localy recently. I got it fixed by locally creating a new conda env with python==3.9.0 . Below link is my requirements.txt file. Hope it will help
https://github.com/WENDGOUNDI/tensorflow_model_maker_production_detection/blob/main/requirements.txt

**Comment by mokshit-pt** [comment]
I found a fix for this by using the model maker on Colab.

Use the command Palette in colab with cmd+shift+P. Then type "use fallback runtime version" and select. This will roll back to Python 3.9.16 runtime and tflite-model-maker can be installed.
Screenshot 2023-05-03 at 12 42 48 PM

Edit: You need to connect to a runtime first, then the option will be available and it will restart with 3.9.16 instead of 3.10.

Also, this will only be available until mid-May so a fix for 3.10 probably is still needed to work with Colab after that.

<img width="636" alt="screenshot" src="https://github.com/tensorflow/tensorflow/assets/106738373/b7a33cdf-7d14-49f0-960e-6eb018cd6d26">

Note: You need to connect to a runtime first, then the option will be available and it will restart with 3.9.16 instead of 3.10.

Reference:
<img width="730" alt="screenshot2" src="https://github.com/tensorflow/tensorflow/assets/106738373/9127bc59-7f9f-4185-97b7-651efe8171c6">

**Comment by AgostinhoSilver** [comment]
> I found a fix for this by using the model maker on Colab.
> 
> Use the command Palette in colab with cmd+shift+P. Then type "use fallback runtime version" and select. This will roll back to Python 3.9.16 runtime and tflite-model-maker can be installed.
> 
> Edit: You need to connect to a runtime first, then the option will be available and it will restart with 3.9.16 instead of 3.10.
> 
> Also, this will only be available until mid-May so a fix for 3.10 probably is still needed to work with Colab after that.
> 
> <img alt="screenshot" width="636" src="https://user-images.githubusercontent.com/106738373/238331362-b7a33cdf-7d14-49f0-960e-6eb018cd6d26.png">
> Note: You need to connect to a runtime first, then the option will be available and it will restart with 3.9.16 instead of 3.10.
> 
> Reference: <img alt="screenshot2" width="730" src="https://user-images.githubusercontent.com/106738373/238331710-9127bc59-7f9f-4185-97b7-651efe8171c6.png">

This no longer working

**Comment by i1Cps** [comment]
This is currently happening with `pip install tf-nightly` , Im on ubuntu
